test/src/test/java/hudson/console/HyperlinkNoteTest.java

```diff
@@ -72,6 +80,28 @@ public void textWithNewlinesModelHyperlinkNote() throws Exception {
                 containsString(new ModelHyperlinkNote("", 0).extraAttributes()),
                 containsString(">" + noteTextSanitized + "</a>")));
     }
+    
+    @Test
+    public void textWithSingleQuote() throws Exception {
+    	FreeStyleProject upstream = r.createFreeStyleProject("upstream");
+    	r.createFreeStyleProject("d0wnstr3'am");
+    	upstream.getPublishersList().add(new BuildTrigger("d0wnstr3'am", Result.SUCCESS));
+    	FreeStyleBuild b = r.buildAndAssertSuccess(upstream);
+        r.waitUntilNoActivity();
+        HtmlPage rsp = r.createWebClient().goTo(b.getUrl()+"console");
+        System.out.println("Before:\n"+rsp.getWebResponse().getContentAsString()+"\n\n");
+        //This would fail if job name has `'`
+        try {
+        	assertThat(String.valueOf(rsp.getAnchorByText("d0wnstr3'am").click().getWebResponse().getStatusCode()), containsString("200"));
```

Comment:
Catching exceptions inside test code is often a sign that the test conditions are not clear.

I believe that you expect this to always succeed.  When it fails, you want the failure message to provide enough information so that the failure can be diagnosed.  Since it is failing in CI, that likely indicates that you should add more assertions to check for additional cases.